### PR TITLE
chore: update CHANGELOG and abi dumps in preparation for release

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6667)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -77,7 +77,62 @@
...
+* feat(storage): round-robin stub for GCS+gRPC (#6584)
+* fix(storage): compile GCS+gRPC with Windows+x86 (#6556)
+* feat: implement unified service account credentials (#6531)
+* feat(storage): install rules for GCS+gRPC plugin (#6527)
+* feat: implement unified insecure credentials (#6518)
+* fix(storage): compile with GCC 11 in C++17 mode (#6501)
+* feat(storage): support service account impersonation (#6488)
+* feat(storage): support method overrides in emulator (#6474)
+* fix(storage): support gzip request body in emulator (#6472)
+* fix(storage): update emulator to use MultipartDecoder (#6453)
```
Comment (coryan) [inline CHANGELOG.md:123]: These 3 are related to the emulator, probably can skip them.

File: CHANGELOG.md [hunk 2]
```diff
@@ -77,7 +77,62 @@
...
+
+* feat(spanner): use multiple channels instead of multiple client for the multiple rows cpu benchmark (#6659)
+* fix(spanner): do not warn about SessionPoolClockOption (#6619)
+
+### Storage
+
+* feat(storage): recommend `g::c::Options` (#6640)
+* feat: custom trust store with unified credentials (#6617)
+* feat(storage): add Retry Test API to emulator (#6611)
+* fix(storage): address behavior bugs (#6574)
```
Comment (coryan) [inline CHANGELOG.md:112]: These two are related to the emulator, probably can skip them.

File: CHANGELOG.md [hunk 3]
```diff
@@ -77,7 +77,62 @@
...
+* fix(storage): compile GCS+gRPC with Windows+x86 (#6556)
+* feat: implement unified service account credentials (#6531)
+* feat(storage): install rules for GCS+gRPC plugin (#6527)
+* feat: implement unified insecure credentials (#6518)
+* fix(storage): compile with GCC 11 in C++17 mode (#6501)
+* feat(storage): support service account impersonation (#6488)
+* feat(storage): support method overrides in emulator (#6474)
+* fix(storage): support gzip request body in emulator (#6472)
+* fix(storage): update emulator to use MultipartDecoder (#6453)
+* feat(storage): impersonate service account credentials (#6441)
```
Comment (coryan) [inline CHANGELOG.md:124]: I think #6441 #6488 #6518 #6531 #6617 can be summarized as:

```md
 * Support unified credentials (`google::cloud::Credentials`) to initialize both the REST and gRPC plugins with the same classes.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -77,7 +77,62 @@
...
+
+### Common Libraries
+
+* feat: custom trust store with unified credentials (#6617)
```
Comment (coryan) [inline CHANGELOG.md:128]: I think you can summarize #6617 #6614 #6531 #6518 as:

```md
 * Add support for "unified" credentials, which allow applications to
   * Configure the REST and gRPC-based libraries using the same credential classes
   * Support complex credential types, such as service account impersonation
   * With this release, only usable in the `storage` library supports these classes.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -77,7 +77,62 @@
...
+* feat(common): make unified Credentials public (#6614)
+* feat: an example using identity tokens with gRPC (#6583)
+* feat: example using identity tokens (#6569)
+* feat: implement unified service account credentials (#6531)
+* feat: implement unified insecure credentials (#6518)
+* feat(common): streaming RPCs that always fail (#6473)
```
Comment (coryan) [inline CHANGELOG.md:134]: I think this and the next are internal-only, we can skip them.

**Review by coryan — APPROVED** [review]
Some ideas to simplify the notes even further.

**Review by dbolduc — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6667)* status: 0 of 7 files reviewed, 7 unresolved discussions (waiting on @scotthart)

___
*[CHANGELOG.md, line 86 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6667#-MbCASFn2UjqB0w4vxwm:-MbCASFn2UjqB0w4vxwn:bq15ckz) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/55dbf35896ace459f8142291796515fa7415593a/CHANGELOG.md#L86)):*
> ```
> * feat(bigtable): add async sample rows (#6561)
> ```

* Add asynchronous method for sampling row keys (#6561). Users that extend `bigtable::DataClient` and wish to use the new method will need to override `PrepareAsyncSampleRowKeys()`.
___
*[CHANGELOG.md, line 87 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6667#-MbCDhMjBR2Nvddm_Hvm:-MbCAFfOEVUiSdAi_TeE:b-xf4rl2) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/55dbf35896ace459f8142291796515fa7415593a/CHANGELOG.md#L87)):*
> ```
> * fix(bigtable): async bulk apply retry, backoff policy fix  (#6510)
> ```

* `bigtable::Table::AsyncBulkApply()` now respects the Retry and Backoff policies (#6510)

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6667)* status: 0 of 7 files reviewed, 7 unresolved discussions (waiting on @coryan, @dbolduc, and @scotthart)

___
*[CHANGELOG.md, line 86 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6667#-MbCASFn2UjqB0w4vxwm:-MbCTCoF2cCUSFXQUjgz:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/55dbf35896ace459f8142291796515fa7415593a/CHANGELOG.md#L86)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

> ```
> * feat(bigtable): add async sample rows (#6561)
> ```

* Add asynchronous method for sampling row keys (#6561). Users that extend `bigtable::DataClient` and wish to use the new method will need to override `PrepareAsyncSampleRowKeys()`.
</blockquote></details>

Done.
___
*[CHANGELOG.md, line 112 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6667#-Mb8_QG-yluSql4L48Wl-r2-112:-MbCT_HW4zeNynIpUUp8:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/55dbf35896ace459f8142291796515fa7415593a/CHANGELOG.md#L112)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

These two are related to the emulator, probably can skip them.
</blockquote></details>

Done.
___
*[CHANGELOG.md, line 123 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6667#-Mb8_QG-yluSql4L48Wl-r2-123:-MbCUD2OBdWvhisoCqih:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/55dbf35896ace459f8142291796515fa7415593a/CHANGELOG.md#L123)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

These 3 are related to the emulator, probably can skip them.
</blockquote></details>

Done.
___
*[CHANGELOG.md, line 124 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6667#-Mb8_QG-yluSql4L48Wl-r2-124:-MbCUDSR5iwLLqmjE7tg:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/55dbf35896ace459f8142291796515fa7415593a/CHANGELOG.md#L124)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

I think #6441 #6488 #6518 #6531 #6617 can be summarized as:

```md
 * Support unified credentials (`google::cloud::Credentials`) to initialize both the REST and gRPC plugins with the same classes.
```

</blockquote></details>

Done.
___
*[CHANGELOG.md, line 128 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6667#-Mb8_QG-yluSql4L48Wl-r2-128:-MbCUnxg0fwu6O4zsxtX:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/55dbf35896ace459f8142291796515fa7415593a/CHANGELOG.md#L128)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

I think you can summarize #6617 #6614 #6531 #6518 as:

```md
 * Add support for "unified" credentials, which allow applications to
   * Configure the REST and gRPC-based libraries using the same credential classes
   * Support complex credential types, such as service account impersonation
   * With this release, only usable in the `storage` library supports these classes.
```

</blockquote></details>

Done.
___
*[CHANGELOG.md, line 134 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6667#-Mb8_QG-yluSql4L48Wl-r2-134:-MbCUoGaBMYmaKHDVFJN:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/55dbf35896ace459f8142291796515fa7415593a/CHANGELOG.md#L134)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

I think this and the next are internal-only, we can skip them.
</blockquote></details>

Done.

File: CHANGELOG.md [hunk 6]
```diff
@@ -101,38 +104,30 @@ experiment with it, but do not ship production code using it yet.
...
+  both the REST and gRPC plugins with the same classes. (#6518)(#6531)(#6617)(#6488)(#6441)
 
 ### Common Libraries
 
-* feat: custom trust store with unified credentials (#6617)
-* feat(common): make unified Credentials public (#6614)
+* feat: Add support for "unified" credentials, which allow applications to
+  Configure the REST and gRPC-based libraries using the same credential classes
+  Support complex credential types, such as service account impersonation
+  With this release, only usable in the `storage` library supports these classes.
```
Comment (coryan) [inline CHANGELOG.md:127]: Ugh, sorry, my grammar was awful here .. maybe "With this release only the `storage` library supports unified credentials"?

**Review by coryan — APPROVED** [review]

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6667)* status: 0 of 7 files reviewed, 8 unresolved discussions (waiting on @coryan, @dbolduc, and @scotthart)

___
*[CHANGELOG.md, line 127 at r3](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6667#-Mb8_QG-yluSql4L48Wl-r3-127:-MbCYM8ACcbJ9GU3N3gg:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/80870bef0709a49e9b596dc2e9ca2f2c75094322/CHANGELOG.md#L127)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Ugh, sorry, my grammar was awful here .. maybe "With this release only the `storage` library supports unified credentials"?
</blockquote></details>

Done.

**Review by dbolduc — APPROVED** [review]

**Pull request merged**